### PR TITLE
update to newest rescript treesitter library and queries

### PR DESCRIPTION
**Pull request by FreddieGilbraith**
The official rescript org tree-sitter repo now includes generated files, helix can depend on it directly.

This MR points to the newest version of the repo, and updates the queries to match the new tree-sitter grammar (coppied from https://github.com/rescript-lang/tree-sitter-rescript/tree/main/queries/rescript)

### CODE REVIEW

File: runtime/queries/rescript/highlights.scm
Comment (the-mikedavis) [inline runtime/queries/rescript/highlights.scm:1]: The new highlights are using neovim capture names like `@number`. We have a custom set of captures here that these will need to be adapted to use: https://docs.helix-editor.com/master/themes.html#syntax-highlighting

File: runtime/queries/rescript/textobjects.scm
Comment (the-mikedavis) [inline runtime/queries/rescript/textobjects.scm:1]: These are also in neovim format. See https://docs.helix-editor.com/master/guides/textobject.html

File: runtime/queries/rescript/highlights.scm
Comment (FreddieGilbraith) [reply runtime/queries/rescript/highlights.scm:1]: TIL that tree-sitter capture names aren't standardized! thanks and sorry, should be fixed now

File: runtime/queries/rescript/textobjects.scm
Comment (FreddieGilbraith) [reply runtime/queries/rescript/textobjects.scm:1]: should now be resolved, as above, thanks

**Review by archseer — DISMISSED** [review]

File: runtime/queries/rescript/locals.scm [hunk 1]
```diff
@@ -1,7 +1,7 @@
 (switch_expression) @local.scope
-(if_expression) @local.scope
 
 ; Definitions
 ;------------
-(type_declaration) @local.defintion
-(let_binding) @local.defintion
+(type_declaration) @local.definition
+(let_binding) @local
+(module_declaration) @local.namespace
```
Comment (the-mikedavis) [inline runtime/queries/rescript/locals.scm:7]:
```suggestion
(let_binding) @local.definition
(module_declaration) @local.definition
```

for locals we only support `local.scope`, `local.definition` and `local.reference` (following https://tree-sitter.github.io/tree-sitter/syntax-highlighting#local-variables)

**Review by the-mikedavis — APPROVED** [review]

**Pull request merged**